Criterion/Measurement.hs

```diff
@@ -313,22 +321,24 @@ measured = Measured {
 -- | Apply the difference between two sets of GC statistics to a
 -- measurement.
 applyGCStatistics :: Maybe GCStatistics
-                  -- ^ Statistics gathered at the __end__ of a run.
+                  -- ^ Statistics gathered at the __end__ of a run, post GC.
+                  -> Maybe GCStatistics
+                  -- ^ Statistics gathered at the __end__ of a run, pre GC.
                   -> Maybe GCStatistics
                   -- ^ Statistics gathered at the __beginning__ of a run.
                   -> Measured
                   -- ^ Value to \"modify\".
                   -> Measured
-applyGCStatistics (Just end) (Just start) m = m {
-    measAllocated          = diff gcStatsBytesAllocated
-  , measNumGcs             = diff gcStatsNumGcs
-  , measBytesCopied        = diff gcStatsBytesCopied
-  , measMutatorWallSeconds = diff gcStatsMutatorWallSeconds
-  , measMutatorCpuSeconds  = diff gcStatsMutatorCpuSeconds
-  , measGcWallSeconds      = diff gcStatsGcWallSeconds
-  , measGcCpuSeconds       = diff gcStatsGcCpuSeconds
-  } where diff f = f end - f start
-applyGCStatistics _ _ m = m
+applyGCStatistics (Just endPostGC) (Just endPreGC) (Just start) m = m {
+    measAllocated          = diff endPostGC start gcStatsBytesAllocated
+  , measNumGcs             = diff endPreGC  start gcStatsNumGcs
+  , measBytesCopied        = diff endPostGC start gcStatsBytesCopied
+  , measMutatorWallSeconds = diff endPostGC start gcStatsMutatorWallSeconds
+  , measMutatorCpuSeconds  = diff endPostGC start gcStatsMutatorCpuSeconds
+  , measGcWallSeconds      = diff endPreGC  start gcStatsGcWallSeconds
+  , measGcCpuSeconds       = diff endPreGC  start gcStatsGcCpuSeconds
```

Comment:
In the comments, can you include a justification for which things are `diff`ed with `endPreGC`, and which things are `diff`ed with `endPostGC`?